aosp_diff/caas/system/sepolicy/0001-fix-for-OTA-update-through-script-failure-1.patch

```diff
@@ -0,0 +1,117 @@
+From fa9bda163d16291ed80ce24757913a2558e0289f Mon Sep 17 00:00:00 2001
+From: "Unnithan, Balakrishnan" <balakrishnan.unnithan@intel.com>
+Date: Wed, 18 Sep 2024 11:41:20 +0530
+Subject: [PATCH] fix for OTA update through script failure-1
+
+OTA update was failing due to sepolicy permission denial.
+
+Added sepolicy to open-read-write block-device block-file.
+
+Tests done: Run command ./update_device.py caas-ota-CR0000094.zip
+
+Tracked-On: OAM-123102
+Signed-off-by: Unnithan, Balakrishnan <balakrishnan.unnithan@intel.com>
+---
+ prebuilts/api/29.0/public/domain.te | 2 +-
+ prebuilts/api/30.0/public/domain.te | 2 +-
+ prebuilts/api/31.0/public/domain.te | 2 +-
+ prebuilts/api/32.0/public/domain.te | 2 +-
+ prebuilts/api/33.0/public/domain.te | 2 +-
+ prebuilts/api/34.0/public/domain.te | 2 +-
+ public/domain.te                    | 2 +-
+ 7 files changed, 7 insertions(+), 7 deletions(-)
+
+diff --git a/prebuilts/api/29.0/public/domain.te b/prebuilts/api/29.0/public/domain.te
+index 1a9e0e1c2..5d1a5bcca 100644
+--- a/prebuilts/api/29.0/public/domain.te
++++ b/prebuilts/api/29.0/public/domain.te
+@@ -430,7 +430,7 @@ neverallow * vendor_init:binder *;
+ 
+ # Don't allow raw read/write/open access to block_device
+ # Rather force a relabel to a more specific type
+-neverallow { domain -kernel -init -recovery } block_device:blk_file { open read write };
++neverallow { domain -kernel -init -recovery -update_engine } block_device:blk_file { open read write };
+ 
+ # Do not allow renaming of block files or character files
+ # Ability to do so can lead to possible use in an exploit chain
+diff --git a/prebuilts/api/30.0/public/domain.te b/prebuilts/api/30.0/public/domain.te
+index c151b9528..c3553dec0 100644
+--- a/prebuilts/api/30.0/public/domain.te
++++ b/prebuilts/api/30.0/public/domain.te
+@@ -428,7 +428,7 @@ neverallow * vendor_init:binder *;
+ 
+ # Don't allow raw read/write/open access to block_device
+ # Rather force a relabel to a more specific type
+-neverallow { domain -kernel -init -recovery } block_device:blk_file { open read write };
++neverallow { domain -kernel -init -recovery -update_engine } block_device:blk_file { open read write };
```

Comment:
which block device need to be accessed by update_engine? you can add a file label for this in file_contexts